src/main/java/io/aiven/kafka/auth/AivenAclAuthorizerV2.java

```diff
@@ -205,13 +210,170 @@ public final List<AuthorizationResult> authorize(final AuthorizableRequestContex
                            action.logIfAllowed(), action.logIfDenied());
 
             final var session = new Session(principal, requestContext.clientAddress());
-            auditor.addActivity(session, action.operation(), action.resourcePattern(), verdict);
+            auditor.addActivity(session, operation, resourcePattern, verdict);
 
             result.add(authResult);
         }
         return result;
     }
 
+    /**
+     * Check if the caller is authorized to perform the given ACL operation on at least one
+     * resource of the given type.
+     *
+     * @param requestContext Request context including request resourceType, security protocol and listener name
+     * @param op             The ACL operation to check
+     * @param resourceType   The resource type to check
+     * @return               Return {@link AuthorizationResult#ALLOWED} if the caller is authorized
+     *                       to perform the given ACL operation on at least one resource of the
+     *                       given type. Return {@link AuthorizationResult#DENIED} otherwise.
+     */
+    @Override
+    public AuthorizationResult authorizeByResourceType(
```

Comment:
The default implementation in `Authorizer` has a sanity checks for the operation and resource type. Also it ensures that superusers are granted access regardless of deny acls. Maybe these should be also respected on the custom implementation?

Reply:
I think concept of superuser is different here, so superuser grant does not work similarly. Same is for normal authorization method, right?

I think I omitted some checks as Kafka itself calls this only for very limited cases. The default implementation only checks for filter types, so if those would be used while calling this implementation, nothing would happen as there is no allow rules for such filter types. So I think this is safe either way.

Reply:
The difference would be great to document here, as I see that the logic is mainly taken from the default implementation.

Reply:
I agree, the differences in the javadoc are not meaningful details here.